api/src/verify.rs

```diff
@@ -72,11 +75,22 @@ pub async fn configured_openid4vp_mdl_request(
 
     let x509c = include_str!("./test/test_event_reader_certificate.b64");
     let x509_bytes = base64::decode(x509c)?;
-    let x509_certificate = x509_certificate::X509Certificate::from_der(x509_bytes)?;
+    let x509_certificate = x509_cert::Certificate::from_der(&x509_bytes)?;
 
     let client_id = x509_certificate
-        .subject_common_name()
-        .context("no client_id in certificate")?;
+        .tbs_certificate
+        .filter::<SubjectAltName>()
+        .filter_map(|r| match r {
+            Ok((_crit, san)) => Some(san.0.into_iter()),
+            Err(_e) => None,
+        })
+        .flatten()
+        .filter_map(|gn| match gn {
+            GeneralName::UniformResourceIdentifier(uri) => Some(uri.to_string()),
+            _ => None,
+        })
+        .collect();
```

Comment:
If there are multiple SAN URI in the certificate then it will concatenate them all into one String. Since you just need one you can use `.next()`